### PR TITLE
Smite cmd

### PR DESCRIPTION
**Pull request by Endermanbugzjfc**
https://user-images.githubusercontent.com/53002741/214257342-261e268e-3254-49ac-b328-f2558ff70743.MOV

# What are missing?
- Damage.
- Fire effect.
- Horrifying message format. (No idea why TextFormat is malfunctioning.)

### CODE REVIEW

**Comment by Endermanbugzjfc** [comment]
# Beyond!
- Smite console.
- Entity selectors. (Colab with [EasyCommandAutoFill](https://poggit.pmmp.io/p/EasyCommandAutofill/3.0.2))

[Screencast from Tuesday, January 24, 2023 06:22:23 PM HKT.webm](https://user-images.githubusercontent.com/53002741/214272719-fd65e1b7-6f75-49fc-86f4-aeec915764c2.webm)

**Comment by Endermanbugzjfc** [comment]
Update: Horrifying message format. is working now.

**Comment by Endermanbugzjfc** [comment]
> # Beyond!
> * Smite console.
> * Entity selectors. (Colab with [EasyCommandAutoFill](https://poggit.pmmp.io/p/EasyCommandAutofill/3.0.2))
> 
>  Screencast.from.Tuesday.January.24.2023.06.22.23.PM.HKT.webm

Note: typo fixed at d7067f0.

**Comment by Endermanbugzjfc** [comment]
# [PHAR direct link](https://poggit.pmmp.io/r/199691/LightningDeath_pr-30.phar)
*If you trust me.*

**Review by DaisukeDaisuke — APPROVED** [review]
LGTM

File: src/Loader.php [hunk 1]
```diff
@@ -2,15 +2,116 @@
...
     public function onEnable() :void{
-        $this->getServer()->getPluginManager()->registerEvents(new LightningListener($this), $this);
+        $this->getServer()->getPluginManager()->registerEvents($this->lightningListener = new LightningListener($this), $this);
         if ($this->getConfig()->get("version") !== 1){
             $this->getLogger()->critical("Please regenerate your config file!");
             $this->getServer()->getPluginManager()->disablePlugin($this);
         }
     }
+    
+    public function onCommand(CommandSender $sender, Command $command, string $alias, array $args) : bool {
```
Comment (provsalt) [inline src/Loader.php:28]: I would move this to a Command class

File: src/Loader.php [hunk 2]
```diff
@@ -2,15 +2,116 @@
...
+            } ?? "";
+        }
+
+        $args = array_unique($args);
+        $countUnknown = 0;
+        foreach ($args as $arg) {
+            if ($arg === "CONSOLE") {
+                // Dear Poggit reviewers: This message is a command output instead of log. Please show some mercy regarding rule B3... Your help is important for such plugin to be "fun".
+                $sender->sendMessage(<<<'EOT'
+                                     .eeeeeeeee
```
Comment (provsalt) [inline src/Loader.php:61]: Unnecessary spam

File: src/Loader.php [hunk 3]
```diff
@@ -2,15 +2,116 @@
...
+                continue;
+            } elseif ($arg === "") {
+                // If @p or @r candidates run out.
+                $countUnknown++;
+                continue;
+            }
+
+            $target = $this->getServer()->getPlayerByPrefix($arg);
+            if ($target === null) {
+               $sender->sendMessage("$arg who?");
```
Comment (provsalt) [inline src/Loader.php:105]: I would prefer if the message would be a proper sentence.

File: src/Loader.php [hunk 4]
```diff
@@ -2,15 +2,116 @@
...
+            if ($target === null) {
+               $sender->sendMessage("$arg who?"); 
+               $countUnknown++;
+            }
+            else $this->lightningListener->Lightning($target);
+        }
+
+        $countArgs = count($args) - $countUnknown;
+        $doThose = $countArgs === 1 ? "does that" : "do those";
+        $sender->sendMessage(TextFormat::BOLD . TextFormat::ITALIC . TextFormat::DARK_RED . "Little $doThose $countArgs know what's gonna happen...");
```
Comment (provsalt) [inline src/Loader.php:113]: Hmm. Is this necessary?

**Review by provsalt — CHANGES_REQUESTED** [review]
There are a few changes that should be changed.

**Pull request closed without merging**